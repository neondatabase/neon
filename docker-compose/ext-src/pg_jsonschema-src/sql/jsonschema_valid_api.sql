-- Define schema
SELECT jsonschema_is_valid('{
  "type": "object",
  "properties": {
    "username": { "type": "string" },
    "age": { "type": "integer" }
  },
  "required": ["username"]
}'::json);

-- Valid instance
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "username": { "type": "string" },
      "age": { "type": "integer" }
    },
    "required": ["username"]
  }'::json,
  '{"username": "alice", "age": 25}'::json
);

-- Invalid instance: missing required "username"
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "username": { "type": "string" },
      "age": { "type": "integer" }
    },
    "required": ["username"]
  }'::json,
  '{"age": 25}'::json
);

-- Invalid instance: wrong type for "age"
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "username": { "type": "string" },
      "age": { "type": "integer" }
    },
    "required": ["username"]
  }'::json,
  '{"username": "bob", "age": "twenty"}'::json
);
