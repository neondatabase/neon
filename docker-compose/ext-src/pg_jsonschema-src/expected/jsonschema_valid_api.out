-- Define schema
SELECT jsonschema_is_valid('{
  "type": "object",
  "properties": {
    "username": { "type": "string" },
    "age": { "type": "integer" }
  },
  "required": ["username"]
}'::json);
 jsonschema_is_valid 
---------------------
 t
(1 row)

-- Valid instance
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "username": { "type": "string" },
      "age": { "type": "integer" }
    },
    "required": ["username"]
  }'::json,
  '{"username": "alice", "age": 25}'::json
);
 jsonschema_validation_errors 
------------------------------
 {}
(1 row)

-- Invalid instance: missing required "username"
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "username": { "type": "string" },
      "age": { "type": "integer" }
    },
    "required": ["username"]
  }'::json,
  '{"age": 25}'::json
);
      jsonschema_validation_errors       
-----------------------------------------
 {"\"username\" is a required property"}
(1 row)

-- Invalid instance: wrong type for "age"
SELECT jsonschema_validation_errors(
  '{
    "type": "object",
    "properties": {
      "username": { "type": "string" },
      "age": { "type": "integer" }
    },
    "required": ["username"]
  }'::json,
  '{"username": "bob", "age": "twenty"}'::json
);
       jsonschema_validation_errors        
-------------------------------------------
 {"\"twenty\" is not of type \"integer\""}
(1 row)

